src/tests/unit-tests.cpp

```diff
@@ -1,6 +1,64 @@
+#include "../app/doctor/doctor.hpp"
+#include "../app/visit/visit.hpp"
 #include "gtest/gtest.h"
 
-TEST(HelloWorld, Hello) {
-    ASSERT_TRUE(true);
+struct AssociationTest : ::testing::Test
+{
+    std::shared_ptr<Doctor> doc1{std::make_shared<Doctor>("Janusz", "Tracz", "1234")};
+    std::shared_ptr<Doctor> doc2{std::make_shared<Doctor>("Lukasz", "Ziobron", "123456")};
+
+    std::shared_ptr<Visit> visit1;
+    std::shared_ptr<Visit> visit2;
+    std::shared_ptr<Visit> visit3;
+
+    AssociationTest()
+    {
+        visit1 = Visit::createVisit(doc1);
+        visit2 = Visit::createVisit(doc2);
+        visit3 = Visit::createVisit(doc2);
+
+        visit2->setVisitInfo("Tworze klub ninja!");
+        visit3->setVisitInfo("I w klubie sa sami fajni ludzie");
+    }
+};
+
+TEST_F(AssociationTest, ExtentDemo)
```

Comment:
Name of the test should say what it does - Demo does not say anything.
Please consider more descriptive naming following some of good test practices for naming, such as:
Given_When_Then
When_Expect

Reply:
It is also very hard to understand, which unit is tested when.
We could use unit division by class or whatever where possible.

Every unit tested should have its own file at least, for example doctor-ut.
whenever we test interaction between classes as whole module/system we need to name files accordingly (I don't know how exactly at the moment).
I think it might be good idea to create seperate folder in tests for unit-tests and other folder/s for combined functionality tests.

Otherwise everything will get messy very soon.